### PR TITLE
Add pnpm support

### PR DESCRIPTION
**Pull request by shahoob**
I just want to use [pnpm](https://pnpm.io) with this cli.

And as a pnpm user, I'd recommend you to use pnpm because it is very efficient on disk space.
Read [this](https://pnpm.io/motivation).

You may request me to migrate from Yarn to pnpm.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "@sapphire/cli",
-	"version": "1.3.1",
+	"version": "1.4.0",
```
Comment (favna) [inline package.json:3]: Revert this. We use automated conventional commit based version bumping.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,16 @@
 
 All notable changes to this project will be documented in this file.
 
+# [1.4.0](https://github.com/sapphiredev/cli/compare/v1.3.1...v1.4.0) - (2022-09-14)
```
Comment (favna) [inline CHANGELOG.md:5]: Revert this. Changelog is auto-generated.

**Review by favna — CHANGES_REQUESTED** [review]
Sapphire currently isn't supported by pnpm because with how it resolves transient dependencies and specifically how @sapphire/pieces isn't put into node_modules/@sapphire/pieces when installing @sapphire/framework. 

This is not something to be resolved on our end but if it can be resolved / configured on pnpm's end then it has to be added to this CLI similar to how for yarn v3 we configure the nodeLinker as node-modules.

File: package.json (hunk 1 above)
Comment (shahoob) [reply package.json:3]: ok

File: CHANGELOG.md (hunk 2 above)
Comment (shahoob) [reply CHANGELOG.md:5]: ok

Comment (shahoob) [reply CHANGELOG.md]: done

File: package.json (hunk 1 above)
Comment (shahoob) [reply package.json]: done

**Comment by shahoob** [comment]
> Sapphire currently isn't supported by pnpm because with how it resolves transient dependencies and specifically how @sapphire/pieces isn't put into node_modules/@sapphire/pieces when installing @sapphire/framework.
> 
> This is not something to be resolved on our end but if it can be resolved / configured on pnpm's end then it has to be added to this CLI similar to how for yarn v3 we configure the nodeLinker as node-modules.

@favna sorry for late comment but pnpm can also hoist all dependencies (when requested)
 
One way to do this is by telling pnpm to `(--)shamefully-hoist` `node_modules` in the `.npmrc` file:
```ini
# pnpm only
shamefully-hoist=true
```
I still don't know if npm and/or Yarn wouldn't mind about this.

You may want to see the [documentation](https://pnpm.io/npmrc#dependency-hoisting-settings) on this.

Although we could also tell pnpm users to remember to put `--shamefully-hoist` in their `pnpm install`/`pnpm add` commands,
humans can make mistakes, so maybe not that way.

**Comment by shahoob** [comment]
I'll create a new PR since I deleted my fork last time

**Comment by shahoob** [comment]
See #191.

**Pull request closed without merging**